single_kernel_mongo/events/database.py

```diff
@@ -66,14 +66,13 @@ def _on_relation_event(self, event: RelationEvent):
         """
         relation_departing = False
         relation_changed = False
-        # TODO : Handle the Mongos VM case
-        # (https://github.com/canonical/mongos-operator/blob/6/edge/lib/charms/mongos/v0/mongos_client_interface.py)
-        # for interface unification.
         if (
             self.dependent.substrate == Substrates.VM
             and self.relation_name == RelationNames.MONGOS_PROXY
         ):
             self.dependent.share_credentials(event.relation)  # type: ignore[attr-defined]
+            return
```

Comment:
Can you explain this return?

Reply:
The client relation is not handler the same way if we are a mongos vm than all other cases (because we are a subordinate charm).

In the very specific case of a mongos vm, we just share credentials to the client application.